File: .claude/plugins/openai-agent-builder/references/project-templates/python-voice-agent/guardrails.py
"""{{PROJECT_NAME}} — Guardrail definitions."""
from pydantic import BaseModel
from agents import (
    Agent,
    GuardrailFunctionOutput,
    RunContextWrapper,
    Runner,
    TResponseInputItem,
    input_guardrail,
)


class ContentCheckOutput(BaseModel):
    is_appropriate: bool
    reasoning: str


guardrail_agent = Agent(
    name="Content checker",
    instructions="Check if the user input is appropriate and on-topic.",
    output_type=ContentCheckOutput,
)


@input_guardrail
async def content_guardrail(
    ctx: RunContextWrapper[None], agent: Agent, input: str | list[TResponseInputItem]
) -> GuardrailFunctionOutput:
    result = await Runner.run(guardrail_agent, input, context=ctx.context)
    return GuardrailFunctionOutput(
        output_info=result.final_output,
        tripwire_triggered=not result.final_output.is_appropriate,
    )

File: .claude/plugins/openai-agent-builder/references/project-templates/typescript-text-agent/src/guardrails.ts
import { Agent, run } from '@openai/agents';
import { z } from 'zod';

const ModerationOutput = z.object({
  isAppropriate: z.boolean(),
  reasoning: z.string(),
});

const guardrailAgent = new Agent({
  name: 'Content checker',
  instructions:
    'Check if the user input is appropriate and on-topic. Return isAppropriate=false for harmful content.',
  outputType: ModerationOutput,
});

export const contentGuardrail = {
  name: 'content_moderation',
  async execute({ input }: { input: string }) {
    const result = await run(guardrailAgent, input);
    return {
      outputInfo: result.finalOutput,
      tripwireTriggered: !result.finalOutput.isAppropriate,
    };
  },
};
